coriolis/tests/osmorphing/test_redhat.py

```diff
@@ -0,0 +1,522 @@
+        self.morphing_tools.disable_predictable_nic_names()
+        mock_exec_cmd_chroot.assert_called_once_with(
+            'grubby --update-kernel=ALL --args="net.ifnames=0 biosdevname=0"')
+
+    @mock.patch.object(
+        redhat.BaseRedHatMorphingTools, '_get_grub2_cfg_location'
+    )
+    def test_get_update_grub2_command(self, mock_get_grub2_cfg_location):
+        result = self.morphing_tools.get_update_grub2_command()
+
+        mock_get_grub2_cfg_location.assert_called_once()
+
+        self.assertEqual(
+            result,
+            'grub2-mkconfig -o %s' % mock_get_grub2_cfg_location.return_value
+        )
+
+    @mock.patch.object(base.BaseLinuxOSMorphingTools, '_exec_cmd_chroot')
+    @mock.patch.object(base.BaseLinuxOSMorphingTools, '_test_path_chroot')
+    def test__get_grub2_cfg_location_uefi(self, mock_test_path_chroot,
+                                          mock_exec_cmd_chroot):
+        mock_test_path_chroot.return_value = True
+
+        result = self.morphing_tools._get_grub2_cfg_location()
+
+        self.assertEqual(result, '/boot/efi/EFI/redhat/grub.cfg')
+        mock_exec_cmd_chroot.assert_has_calls([
+            mock.call("mount /boot || true"),
+            mock.call("mount /boot/efi || true")
+        ])
+        mock_test_path_chroot.assert_called_once_with(
+            '/boot/efi/EFI/redhat/grub.cfg')
+
+    @mock.patch.object(base.BaseLinuxOSMorphingTools, '_exec_cmd_chroot')
+    @mock.patch.object(base.BaseLinuxOSMorphingTools, '_test_path_chroot')
+    def test__get_grub2_cfg_location_bios(self, mock_test_path_chroot,
+                                          mock_exec_cmd_chroot):
+        mock_test_path_chroot.side_effect = [False, True]
+
+        result = self.morphing_tools._get_grub2_cfg_location()
+
+        mock_test_path_chroot.assert_called_with('/boot/grub2/grub.cfg')
+        mock_exec_cmd_chroot.assert_has_calls([
+            mock.call("mount /boot || true"),
+            mock.call("mount /boot/efi || true")
+        ])
+
+        self.assertEqual(result, '/boot/grub2/grub.cfg')
+
+    @mock.patch.object(base.BaseLinuxOSMorphingTools, '_exec_cmd_chroot')
+    @mock.patch.object(base.BaseLinuxOSMorphingTools, '_test_path_chroot')
+    def test__get_grub2_cfg_location_unknown(self, mock_test_path_chroot,
+                                             mock_exec_cmd_chroot):
+        mock_test_path_chroot.return_value = False
+
+        self.assertRaisesRegex(
+            Exception,
+            "could not determine grub location. boot partition not mounted?",
+            self.morphing_tools._get_grub2_cfg_location
+        )
+
+        mock_exec_cmd_chroot.assert_has_calls([
+            mock.call("mount /boot || true"),
+            mock.call("mount /boot/efi || true")
+        ])
+        mock_test_path_chroot.assert_has_calls([
+            mock.call('/boot/efi/EFI/redhat/grub.cfg'),
+            mock.call('/boot/grub2/grub.cfg')
+        ])
+
+    @ddt.data(
```

Comment:
Please use this formatting for the `@ddt.data` inputs:
```python
@ddt.data(
    (
        [("/path/to/ifcfg-file", {"HWADDR": "mac", "NAME": "test-name"})],
        [],
        [("test-name", "mac")]
    ),
    (
        [("/path/to/ifcfg-file", {"NAME": "test-name"})],
        ["mac"],
        [("test-name", "mac")]
    ),
    (
        [("/path/to/ifcfg-file", {"HWADDR": "mac"})],
        ["mac"],
        [("file", "mac")]
    ),
    (
        [("/path/to/ifcfg-file", {"NAME": "test-name"})],
        [],
        []
    )
)
```

this way, it's more clear which what values map to `ifcfgs_ethernet`, `mac_addresses`, or `expected`.